File: src/main/java/ro/mpp2024/Validator/ParticipantValidator.java
package ro.mpp2024.Validator;

import ro.mpp2024.Domain.Participant;

public class ParticipantValidator implements AbstractValidator<Participant> {
    @Override
    public void validate(Participant participant) {
        String errors = "";
        if(participant.getNume().isEmpty())
            errors += "Nume invalid!\n";
        if(participant.getPrenume().isEmpty())
            errors += "Prenume invalid!\n";
        if(participant.getVarsta() < 18)
            errors += "Varsta invalida!\n";
        if(!errors.isEmpty())
            throw new RuntimeException(errors);
    }
}

File: SwimmingCompetitionCSharpGUI/Validator/ParticipantValidator.cs
using SwimmingCompetition.Domain;

namespace SwimmingCompetition.Validator;

public class ParticipantValidator: IAbstractValidator<Participant>
{
    public void Validate(Participant e)
    {
        String errors = "";

        if (String.IsNullOrEmpty(e.Nume))
            errors += "Nume invalid!\n";
        if (String.IsNullOrEmpty(e.Prenume))
            errors += "Prenume invalid!\n";
        if (e.Varsta < 18)
            errors += "Varsta invalida!\n";
        
        if (String.IsNullOrEmpty(errors))
            throw new Exception(errors);
    }
}